pandas/tests/arrays/categorical/test_dtypes.py

```diff
@@ -176,6 +178,13 @@ def test_astype_category(self, dtype_ordered, cat_ordered):
             expected = cat
             tm.assert_categorical_equal(result, expected)
 
+    def test_astype_object_datetime_categories(self):
+        # GH#40754
+        cat = Categorical(to_datetime(["2021-03-27", NaT]))
+        result = cat.astype(object)
+        expected = np.array([Timestamp("2021-03-27 00:00:00"), np.nan], dtype="object")
```

Comment:
why doesn't this have NaT?

Reply:
Because of 

```
fill_value = lib.item_from_zerodim(np.array(np.nan).astype(dtype))
```

in the astype call. Could change that theoretically. But would prefer doing this in a follow up if necessary

Reply:
yah, i think using fill_value=self.categories._na_value in categorical L549 is the way to go.  also i think do `new_cats = self.categories._values` instead of L542-545.  follow-up is reasonable.

Reply:
Will do as a follow up then after this is merged